src/profile_service/index.js

```diff
@@ -79,12 +79,17 @@ function getRSASignatureForMessage(message, pem) {
 }
 
 function decryptCurrentUserReceipt(receipt, pem, callback) {
-  let unwrappedKey = unwrapKey(receipt.wrapped_receipt_key, pem);
-  let decodedData = protoRoot.decodeEncryptedData(new Buffer(receipt.other_party_profile_content, 'base64'))
-  let iv = forge.util.decode64(decodedData.iv);
-  let cipherText = forge.util.decode64(decodedData.cipherText);
-
-  return decipherProfile(cipherText, forge.util.decode64(unwrappedKey), iv);
+  if(receipt.other_party_profile_content && Object.keys(receipt.other_party_profile_content).length > 0) {
```

Comment:
I usually replace the if-else block with guard clause:

```javascript
function decryptCurrentUserReceipt(receipt, pem, callback) {
  if (!receipt.other_party_profile_content || Object.keys(receipt.other_party_profile_content).length === 0) {
    console.log('no decrypted data')
    return []
  }

  const unwrappedKey = unwrapKey(receipt.wrapped_receipt_key, pem);
  const decodedData = protoRoot.decodeEncryptedData(new Buffer(receipt.other_party_profile_content, 'base64'))
  const iv = forge.util.decode64(decodedData.iv);
  const cipherText = forge.util.decode64(decodedData.cipherText);

  return decipherProfile(cipherText, forge.util.decode64(unwrappedKey), iv);
}
```
Works either way, so you can ignore this.